packages/react-composites/src/composites/ChatComposite/adapter/ChatAdapter.ts

```diff
@@ -87,6 +103,18 @@ export interface ChatAdapterThreadManagement {
    *
    */
   loadPreviousChatMessages(messagesToLoad: number): Promise<boolean>;
+  /* @conditional-compile-remove-from(stable): FILE_SHARING */
+  /**
+   * Sets the uploaded files in ui state.
+   * @beta
+   */
+  uploadFiles?: (uploadedFiles: UploadedFile[]) => void;
```

Comment:
This method name is confusing. The adapter doesn't take any action to upload files. Should be more like `registerActiveFileUpload`.

The current setup isn't quite right -- we can't inject mutable objects into the adapter state. The adapter state is an immutable refrentially stable object.

See how `CallComposite` deals with `isLocalMicrophoneEnabled` -- `setIsLocalMicrophoneEnabled` is exposed as the only way to modify this field, and it uses `immer` to do it correctly.

Reply:
One possibility is that you add a method `registerActiveFileUpload` that *returns* an object with the interface passed to the Contoso callback. Internally, the adapter Context subscribes to events on this object to immutably update the adapter (ui) state.

Reply:
I implemented `setIsLocalMicrophoneEnabled` :D and even I don't remember it. Haha